idea/src/org/jetbrains/kotlin/idea/intentions/ConvertToForEachFunctionCallIntention.kt

```diff
@@ -35,15 +34,45 @@ class ConvertToForEachFunctionCallIntention : SelfTargetingIntention<KtForExpres
     override fun applyTo(element: KtForExpression, editor: Editor?) {
         val commentSaver = CommentSaver(element)
 
+        val labelName = element.getLabelName()
+
         val body = element.body!!
         val loopParameter = element.loopParameter!!
 
         val functionBodyArgument: Any = if (body is KtBlockExpression) body.contentRange() else body
 
-        val foreachExpression = KtPsiFactory(element).createExpressionByPattern(
+        val psiFactory = KtPsiFactory(element)
+        val foreachExpression = psiFactory.createExpressionByPattern(
                 "$0.forEach{$1->$2}", element.loopRange!!, loopParameter, functionBodyArgument)
-        val result = element.replace(foreachExpression)
+        val result = element.replace(foreachExpression) as KtElement
+
+        result.getContinuesWithLabel(labelName).forEach {
+            it.replace(psiFactory.createExpression("return@forEach"))
+        }
 
         commentSaver.restore(result)
     }
+
+    private fun KtElement.getContinuesWithLabel(labelName: String?): List<KtContinueExpression> {
+        val continueElements = ArrayList<KtContinueExpression>()
+
+        forEachDescendantOfType<KtContinueExpression>({ it !is KtLoopExpression }) {
```

Comment:
OK, agreed regarding the two loops. As for the second comment, intentions are not always invoked on green code, and it's usually best to avoid breaking it even further.